python/lsst/ts/MTAlignment/alignmentModel.py

```diff
@@ -0,0 +1,507 @@
+            elif stat == "ERR":
+                self.trackerstatus = TrackerStatus.ERR
+            await asyncio.sleep(1)
+
+    async def wait_for_ready(self):
+        """
+        Checks if the tracker is doing something, if so, sleep until it's done.
+        """
+        with self.com_lock:
+            wait_states = ("EMP\r\n")
+            msg = bytes("?STAT\r\n", 'ascii')
+            self.writer.write(msg)
+            await self. writer.drain()
+            data = await  self.reader.read(64)
+            stat = data.decode()
+            if stat == "INIT" or stat == "INIT\r\n":
+                self.log.debug("waiting for init")
+                await asyncio.sleep(5)
+            while stat in wait_states:
+                self.log.debug("...")
+                await asyncio.sleep(0.3)
+                self.writer.write(msg)
+                await self. writer.drain()
+                data = await  self.reader.read(64)
+                stat = data.decode()
+            await asyncio.sleep(0.5)
+
+    async def send_msg(self, msg):
+        """
+        Formats and sends a message to T2SA.
+        """
+        msg = msg + "\r\n"
+        if type(msg) == str:  # this may move
+            msg = bytes(msg, 'ascii')
+        self.log.debug(f"sending {msg}")
+        with self.com_lock:
+            self.writer.write(msg)
+            await self. writer.drain()
+            data = await self.reader.readuntil(separator=bytes("\n", 'ascii'))
+            self.log.debug(f'Received: {data.decode()!r}')
+            return data.decode()
+
+    async def status(self):
+        """
+        query T2SA for status
+        """
+        response = await self.send_msg("?STAT")
+        return response
+
+    async def laser_status(self):
+        """
+        Query tracker laser status, and update model state accordingly
+        """
+
+        msg = self.msgformat("?LSTA")
+        data = await self.send_msg(msg)
+        if data == "LNC":
+            self.laserstatus = LaserStatus.LASERNOTCONNECTED
+        elif data == "LOFF":
+            self.laserstatus = LaserStatus.LASEROFF
+        elif data == "LON":
+            self.laserstatus = LaserStatus.LASERON
+        elif data[0:4] == "WARM":
+            self.laserstatus = LaserStatus.WARMING
+            # TODO do something with the warmup timer in data[:4]
+
+    async def laser_on(self):
+        """
+        Turns the Tracker Laser on (for warmup purposes)
+        """
+
+        msg = "!LST:1"
+        data = await self.send_msg(msg)
+        return data
+
+    async def laser_off(self):
+        """
+        Turns the Tracker Laser off
+        """
+
+        msg = "!LST:0"
+        data = await self.send_msg(msg)
+        return data
+
+    async def measure_m2(self):
+        """
+        Execute M2 measurement plan
+        """
+
+        if self.first_measurement:
+            self.log.debug("first measurement, waiting for tracker warmup")
+            await asyncio.sleep(5)
```

Comment:
I think it would be good if the sleep time here was an attribute of the class, e.g.;

```
await asyncio.sleep(self.warmup_time)
```